### PR TITLE
Support libgbinder for Android 11

### PR DESCRIPTION
**Pull request by mlehtima**
This supersedes https://github.com/mer-hybris/libgbinder/pull/75, code has been rebased, style issues and commit messages fixed and missing and broken unit tests added/fixed.

### CODE REVIEW

**Comment by madushan1000** [comment]
Hi @mlehtima , can you integrate the changes from here(https://github.com/adglkh/libgbinder/pull/1/files) too? the service managed protocol has a "stability" byte now. The old get_service will not work for aidl3.

**Comment by mlehtima** [comment]
@madushan1000  I added the get_service code from your pull request, it seemed to work correctly now.

**Comment by madushan1000** [comment]
Nice, how are you testing these btw? with a real android device?

**Comment by mlehtima** [comment]
Testing is done on a Sailfish OS device with adaptation side built against Android 11 base.

File: src/gbinder_servicemanager_aidl3.c [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+
+        GBinderReader reader;
+        gbinder_remote_reply_init_reader(reply, &reader);
+        /* Read status */
+        GVERIFY(gbinder_reader_read_int32(&reader, &status));
+        /* Read size */
+        int srv_size = 0;
+        GVERIFY(gbinder_reader_read_int32(&reader, &srv_size));
+        /* Iterate each service name */
+        for (int i = 0; i < srv_size; i++) {
```
Comment (monich) [inline src/gbinder_servicemanager_aidl3.c:157]:
```
src/gbinder_servicemanager_aidl3.c: In function ‘gbinder_servicemanager_aidl3_list’:
src/gbinder_servicemanager_aidl3.c:157:9: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
         for (int i = 0; i < srv_size; i++) {
```

Comment (mlehtima) [reply src/gbinder_servicemanager_aidl3.c]: Fixed

**Review by monich — APPROVED** [review]

**Pull request merged**